src/main/kotlin/com/example/passmanager/repositories/impl/PassRepositoryImpl.kt

```diff
@@ -0,0 +1,124 @@
+        val pipelineStages = getPassesLookUpAggregationPipeline(passOwnerId).operations
+        val aggregationResults =
+            mongoTemplate.aggregate<ProjectionResult>(
+                newAggregation(pipelineStages),
+                PASS_OWNER_COLLECTION
+            )
+        return aggregationResults.uniqueMappedResult?.passes.orEmpty()
+    }
+
+    override fun findById(passId: String): MongoPass? {
+        return mongoTemplate.findById<MongoPass>(passId)
+    }
+
+    override fun insert(newPass: MongoPass): MongoPass {
+        return mongoTemplate.insert(newPass)
+    }
+
+    override fun save(pass: MongoPass): MongoPass {
+        return mongoTemplate.save(pass)
+    }
+
+    override fun deleteById(passId: String) {
+        mongoTemplate.remove<MongoPass>(query(where("_id").isEqualTo(passId)))
+    }
+
+    override fun deleteAllByOwnerId(passOwnerId: String) {
+        mongoTemplate.remove<MongoPass>(query(where("passOwnerId").isEqualTo(ObjectId(passOwnerId))))
+    }
+
+    override fun deleteByIdAndOwnerId(passId: String, passOwnerId: String): Boolean {
+        val deleteResult = mongoTemplate.remove<MongoPass>(
+            query(where("passOwnerId").isEqualTo(ObjectId(passOwnerId)).andOperator(where("_id").isEqualTo(passId)))
+        )
+        return deleteResult.deletedCount > 0
+    }
+
+    override fun sumPurchasedAtAfterDate(passOwnerId: String, afterDate: LocalDate): BigDecimal {
+        val aggregation = newAggregation(
+            getPassesByOwnerAndPurchasedAfterPipeline(passOwnerId, afterDate)
+                .add(group().sum("purchasedFor").`as`("total"))
+                .add(project("total").andExclude("_id"))
+                .operations
+        )
+        val aggregationResults = mongoTemplate.aggregate<SumResult>(aggregation, COLLECTION_NAME)
+        return aggregationResults.uniqueMappedResult?.total ?: BigDecimal.ZERO
+    }
+
+    override fun getPassesPriceDistribution(passOwnerId: String): List<PriceDistribution> {
+        val aggregation = newAggregation(
+            match(where("passOwnerId").isEqualTo(ObjectId(passOwnerId))),
+            lookup().from(PASS_TYPE_COLLECTION)
+                .localField("passTypeId")
+                .foreignField("_id")
+                .`as`("types"),
+            unwind("types"),
+            group("types.name").sum("types.price").`as`("spentForPassType"),
+            project("spentForPassType").and("_id").`as`("typeName").andExclude("_id")
+        )
+        val aggregationResults = mongoTemplate.aggregate<PriceDistribution>(aggregation, COLLECTION_NAME)
+        return aggregationResults.mappedResults
+    }
+
+    private fun getPassesByOwnerAndPurchasedAfterPipeline(passOwnerId: String, date: LocalDate): AggregationPipeline {
+        return AggregationPipeline.of(
+            match(where("passOwnerId").isEqualTo(ObjectId(passOwnerId)).andOperator(where("purchasedAt").gt(date))),
```

Comment:
Looks like you compare `Instant` (mongo model) and `LocalDate` here

Reply:
Yeah, I wanted to use `gte` instead of `gt.` Idea is to get all passes after particular date (including this date).

LocalDate is taken from dto input, and compares from Instant model field, so if, for example, pass has `purchasedAt: 2024-09-20T07:19:13.000+00:00`,  so for DTO request date: `2024-09-20` it will satisfy condition